gandalf.js

```diff
@@ -1,26 +1,28 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.